src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -337,6 +348,16 @@ public Task WriteStreamAsync(Stream stream, CancellationToken cancellationToken
                 }
             }
         }
+
+        internal class TemporaryStorageTextReader : TextReader
```

Comment:
There only appears to be one subclass of this type.  Why not just roll this into DirectMemoryAccessReader?